generators/app/templates/src/app/core/http/cache.interceptor.ts

```diff
@@ -36,15 +36,36 @@ export class CacheInterceptor implements HttpInterceptor {
     return new Observable((subscriber: Subscriber<HttpEvent<any>>) => {
       const cachedData = this.forceUpdate ? null : this.httpCacheService.getCacheData(request.urlWithParams);
       if (cachedData !== null) {
-        // Create new response to avoid side-effects
-        subscriber.next(new HttpResponse(cachedData as Object));
+        // Create new HttpResponse to be restored
+        const data = {
+          body: cachedData.body,
+          headers: new HttpHeaders({ ...cachedData.headers }),
+          status: cachedData.status,
+          statusText: cachedData.statusText,
+          url: cachedData.url
+        };
+        subscriber.next(new HttpResponse(data));
         subscriber.complete();
       } else {
         next.handle(request)
           .subscribe(
             event => {
               if (event instanceof HttpResponse) {
-                this.httpCacheService.setCacheData(request.urlWithParams, event);
+                // Convert HttpHeaders in an object
+                const headers = event.headers
+                  .keys()
+                  .reduce((accumulator: any, key: any) => ({ ...accumulator, [key]: event.headers.get(key) }), {});
+
+                // Create a new HttpResponse to be cached
+                const data = {
+                  body: event.body,
+                  headers: headers,
+                  status: event.status,
+                  statusText: event.statusText,
+                  url: event.url
+                } as HttpResponse<any>;
```

Comment:
instead of tricking the type system, it would be better to create a simple interface for this instead, like `HttpResponseData` and use it instead.

Reply:
I quite neebie with ng 7 and typescript (only 3 weeks), I don't know why, by every tests here is broken. I tried to create this new interface `HttpResponseData` as you suggested but to convert the `HttpResponse` to `HttpResponseData` a tricky type conversion is necessary and also the methods `setCacheData` and `getCacheData` of `HttpCacheService` accept the type `HttpResponse` so here also I should change these types also `HttpCacheEntry`.

Reply:
Yes, I may have omitted but for this to work of course you also have to use `HttpResponseData` in `HttpCacheService` methods and types 👍